docs/software-architecture-and-design.md

```diff
@@ -472,6 +475,30 @@ The most important thing to know about editing the [Visualizer](#visualizer) in
 
 More information about defining custom widgets in [QtCreator](https://doc.qt.io/qtcreator/creator-using-qt-designer.html) can be found [here](https://doc.qt.io/qt-5/designer-using-custom-widgets.html).
 
+## Qt Best Practices
+Although this will focus on Qt-specific examples, these principles generally apply to all GUI design and implementation. They are really just Software Engineering principles applied to GUIs.
+
```

Comment:
Maybe we should have a naming convention/guide for elements that interact with code, e.g. radioButton_ChargeMode

Reply:
Mmm good idea. I'll make an issue to look into this. We'll likely also want to experiment with the link between `ui` defined stuff and code, as well as nesting `ui` files.

Reply:
Made #1309 for now